src/H5Dchunk.c

```diff
@@ -8260,70 +8262,11 @@ H5D__chunk_get_offset_copy(const H5D_t *dset, const hsize_t *offset, hsize_t *of
         /* Make sure the offset fall right on a chunk's boundary */
         if (offset[u] % dset->shared->layout.u.chunk.dim[u])
             HGOTO_ERROR(H5E_DATASET, H5E_BADVALUE, FAIL, "offset doesn't fall on chunks's boundary");
-
-        offset_copy[u] = offset[u];
     }
 
 done:
     FUNC_LEAVE_NOAPI(ret_value)
-} /* end H5D__chunk_get_offset_copy() */
-
-/*-------------------------------------------------------------------------
- * Function:    H5D__write_struct_chunk_direct
- *
- * Purpose:    Internal routine to write a structured chunk directly into the file.
- *
- * Return:    Non-negative on success/Negative on failure
- *
- *-------------------------------------------------------------------------
- */
-herr_t
-H5D__write_struct_chunk_direct(H5D_t H5_ATTR_UNUSED *dset, hsize_t H5_ATTR_UNUSED *offset,
```

Comment:
Unused at this point or were these moved elsewhere?

Reply:
Unused, should not be in H5Dchunk.c, and should be implemented in terms of H5SC layout callbacks. The H5SC code will handle the actual direct I/O, the only thing the layout callbacks need to do is insert the chunk into the index.